packages/expo-cli/src/commands/build-native/iOSBuilder.ts

```diff
@@ -49,14 +56,14 @@ class iOSBuilder implements Builder {
     if (!this.shouldLoadCredentials()) {
       return;
     }
-    const bundleIdentifier = this.ctx.exp?.ios?.bundleIdentifier;
-    if (!bundleIdentifier) {
+    this.bundleIdentifier = this.ctx.exp?.ios?.bundleIdentifier;
```

Comment:
this won't work for builds that does not require credentials because this method will exit at the begining

as a side note I think that in general, it's better to duplicate those 4 lines of code that to create class member to hold it between the calls